cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/servlet/handler/sync/AdminRequest.java

```diff
@@ -0,0 +1,141 @@
+
+    Map<AnomalyType, Boolean>
+        selfHealingBefore = new HashMap<>(disableSelfHealingFor.size() + enableSelfHealingFor.size());
+    Map<AnomalyType, Boolean> selfHealingAfter = new HashMap<>(disableSelfHealingFor.size() + enableSelfHealingFor.size());
+
+    for (AnomalyType anomalyType : disableSelfHealingFor) {
+      selfHealingBefore.put(anomalyType, _kafkaCruiseControl.setSelfHealingFor(anomalyType, false));
+      selfHealingAfter.put(anomalyType, false);
+    }
+
+    for (AnomalyType anomalyType : enableSelfHealingFor) {
+      selfHealingBefore.put(anomalyType, _kafkaCruiseControl.setSelfHealingFor(anomalyType, true));
+      selfHealingAfter.put(anomalyType, true);
+    }
+
+    if (!disableSelfHealingFor.isEmpty() || !enableSelfHealingFor.isEmpty()) {
+      LOG.warn("Self healing state is modified by user (before: {} after: {}).", selfHealingBefore, selfHealingAfter);
+    }
+
+    // 3. Drop selected recently removed/demoted brokers.
+    String dropRecentBrokersRequest = processDropRecentBrokersRequest();
+
+    return new AdminResult(selfHealingBefore,
+                           selfHealingAfter,
+                           ongoingConcurrencyChangeRequest.isEmpty() ? null : ongoingConcurrencyChangeRequest,
+                           dropRecentBrokersRequest,
+                           _kafkaCruiseControl.config());
+  }
+
+  private String processDropRecentBrokersRequest() {
+    StringBuilder sb = new StringBuilder();
+
+    Set<Integer> brokersToDropFromRecentlyRemoved = _parameters.dropRecentlyRemovedBrokers();
+    if (!brokersToDropFromRecentlyRemoved.isEmpty()) {
+      if (!_kafkaCruiseControl.dropRecentBrokers(brokersToDropFromRecentlyRemoved, true)) {
+        Set<Integer> recentlyRemovedBrokers = _kafkaCruiseControl.recentBrokers(true);
+        sb.append(String.format("None of the brokers to drop (%s) are in the recently removed broker set"
+                                + " (%s).%n", brokersToDropFromRecentlyRemoved, recentlyRemovedBrokers));
+        LOG.warn("None of the user-requested brokers to drop ({}) are in the recently removed broker set ({}).",
+                 brokersToDropFromRecentlyRemoved, recentlyRemovedBrokers);
+      } else {
+        Set<Integer> recentlyRemovedBrokers = _kafkaCruiseControl.recentBrokers(true);
+        sb.append(String.format("Dropped recently removed brokers (requested: %s after-dropping: %s).%n",
+                                brokersToDropFromRecentlyRemoved, recentlyRemovedBrokers));
+        LOG.warn("Recently removed brokers are dropped by user (requested: {} after-dropping: {}).",
+                 brokersToDropFromRecentlyRemoved, recentlyRemovedBrokers);
+      }
+    }
+
+    Set<Integer> brokersToDropFromRecentlyDemoted = _parameters.dropRecentlyDemotedBrokers();
+    if (!brokersToDropFromRecentlyDemoted.isEmpty()) {
+      if (!_kafkaCruiseControl.dropRecentBrokers(brokersToDropFromRecentlyRemoved, false)) {
```

Comment:
`brokersToDropFromRecentlyRemoved` -> `brokersToDropFromRecentlyRemoved`

Reply:
Ah yes it should be `...` -> `brokersToDropFromRecentlyDemoted`.